lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -3066,6 +3066,12 @@ bool LowererMDArch::GenerateFastDivAndRem(IR::Instr* instrDiv, IR::LabelInstr* b
     IR::Opnd* divisor   = instrDiv->GetSrc2(); // denominator
     IR::Opnd* dst       = instrDiv->GetDst();
 
+    if(static_cast<IR::RegOpnd*>(divident)->IsSameRegUntyped(dst) && (instrDiv->m_opcode == Js::OpCode::Rem_I4 || instrDiv->m_opcode == Js::OpCode::RemU_I4 || bailOutLabel))
+    {
+        divident = IR::RegOpnd::New(TyInt32, instrDiv->m_func);
```

Comment:
should probably use instrDiv->GetSrc1()->GetSize() rather than hard coded int32

Reply:
The current optimization is for i32 only. will modify to prevent other types from reaching this path.